addons/purchase_stock/models/purchase.py

```diff
@@ -284,18 +284,21 @@ def _create_or_update_picking(self):
     @api.multi
     def _get_stock_move_price_unit(self):
         self.ensure_one()
-        line = self[0]
-        order = line.order_id
-        price_unit = line.price_unit
-        if line.taxes_id:
-            price_unit = line.taxes_id.with_context(round=False).compute_all(
-                price_unit, currency=line.order_id.currency_id, quantity=1.0, product=line.product_id, partner=line.order_id.partner_id
+        vals = self._prepare_compute_all_values()
+        order = self.order_id
+        price_unit = vals['price_unit']
+        if self.taxes_id:
+            price_unit = self.taxes_id.with_context(round=False).compute_all(
+                price_unit, currency=vals['currency_id'],
+                quantity=vals['product_qty'], product=vals['product'],
```

Comment:
I have put here the quantity of the purchase line, but originally there was `1.0` as fixed value. It should change anything, but it's worth to be noted if you think it should be reverted to the fixed value.